### ISSUE TITLE
Separate lib/pre-binding.js into its own npm module?

### ISSUE THREAD

**Issue by bminer** [mapbox/node-pre-gyp#191]
This is just a thought, so please bear with me here...

node-pre-gyp's dependencies have rather large file sizes (i.e. request weighs 2.6MB), and node-pre-gyp weights over 4MB!  For this reason, I'd like to bundle my pre-compiled module into a nice, small tarball that doesn't need node-pre-gyp as a dependency; however, I like that node-pre-gyp replaces the npm bindings module with the following code:

```javascript
var binary = require('node-pre-gyp');
var path = require('path');
var binding_path = binary.find(path.resolve(path.join(__dirname,'./package.json')));
var binding = require(binding_path);
```

Is there any way that the functionality of `lib/pre-binding.js` could be separated into another npm module that is required by node-pre-gyp?  That way, I could easily delete node-pre-gyp as a dependency after building/installation and still have the binding path determined dynamically.

Does that make sense?  Thoughts on this?

**Comment by bminer** [comment]
Any thoughts on this issue?

**Comment by addaleax** [comment]
+1 from me for this! And it does seem quite feasible to me, too.

**Comment by bminer** [comment]
Would you guys consider a PR for this change?

**Comment by springmeyer** [comment]
Yes, this would be good to do. I started this at https://github.com/mapbox/node-pre-gyp/pull/80. Will put on my list to refresh that effort.

**Comment by jeroenvollenbrock** [comment]
Is there any progress on this? I do not like the fact that npmlog is included upon runtime, as it includes gauge which uses signal-exit to hijack process signals in a way that is not compatible with similar modules. Therefore our application stops responding to ctrl+c or SIGINTs as soon as a module using node-pre-gyp gets included.
